sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -51,21 +54,29 @@ export class ConfirmationOfEnrollmentController {
    * @returns student application list of an institution location
    */
   @HasLocationAccess("locationId")
-  @Get(":locationId/confirmation-of-enrollment")
+  @Get(
+    ":locationId/confirmation-of-enrollment/enrollmentPeriod/:enrollmentPeriod",
+  )
   async getCOESummary(
     @Param("locationId") locationId: number,
+    @Param("enrollmentPeriod") enrollmentPeriod: string,
   ): Promise<COESummaryDTO[]> {
-    const applications = await this.applicationService.getCOEApplications(
-      locationId,
-    );
-    return applications.map((eachApplication: Application) => {
+    const disbursementSchedules =
+      await this.disbursementScheduleService.getCOEByLocation(
+        locationId,
+        enrollmentPeriod === EnrollmentPeriod.Upcoming,
+      );
+    return disbursementSchedules.map((disbursement: DisbursementSchedule) => {
       return {
-        applicationNumber: eachApplication.applicationNumber,
-        applicationId: eachApplication.id,
-        studyStartPeriod: eachApplication.offering?.studyStartDate ?? "",
-        studyEndPeriod: eachApplication.offering?.studyEndDate ?? "",
-        coeStatus: eachApplication.coeStatus,
-        fullName: getUserFullName(eachApplication.student.user),
+        applicationNumber: disbursement.application.applicationNumber,
+        applicationId: disbursement.application.id,
+        studyStartPeriod:
```

Comment:
Is there a reason to expect that COE will not have studyStartPeriod and studyEndPeriod. Maybe these null checks are from the past but I do not see a reason why the COE would not have offering dates. Please let m know if I am missing something.

Reply:
Makes sense. Offering and these dates can never be NULL.

Reply:
True, at this part of the workflow, the start and end period cannot be null.

Reply:
NULL checks removed.